lib/enkf/enkf_main.cpp

```diff
@@ -1107,31 +1104,97 @@ static void enkf_main_analysis_update( enkf_main_type * enkf_main ,
       if (local_dataset_get_size( dataset )) {
         local_obsdata_type   * local_obsdata = local_ministep_get_obsdata( ministep );
 
-        // The enkf_main_serialize_dataset() function will query the storage
-        // layer and fetch data which is serialized into the A matrix which is
-        // buried deep into the serialize_info structure.
         enkf_main_serialize_dataset(enkf_main_get_ensemble_config(enkf_main), dataset , step2 ,  use_count , tp , serialize_info);
         module_info_type * module_info = enkf_main_module_info_alloc(ministep, obs_data, dataset, local_obsdata, serialize_info->active_size.data() , serialize_info->row_offset.data());
 
-        if (analysis_module_check_option( module , ANALYSIS_UPDATE_A)){
-          if (analysis_module_check_option( module , ANALYSIS_ITERABLE)){
-            analysis_module_updateA( module , localA , S , R , dObs , E , D , module_info, enkf_main->shared_rng);
-          }
-          else
-            analysis_module_updateA( module , localA , S , R , dObs , E , D , module_info, enkf_main->shared_rng);
-        } else {
-          if (analysis_module_check_option( module , ANALYSIS_USE_A)){
-            analysis_module_initX( module , X , localA , S , R , dObs , E , D, enkf_main->shared_rng);
+        /*
+          The update for one local_dataset instance consists of two main chunks:
+
+          1. The first chunk updates all the parameters which don't have row
+             scaling attached. These parameters are serialized together to the A
+             matrix and all the parameters are updated in one go.
+
+          2. The second chunk is loop over all the parameters which have row
+             scaling attached. These parameters are updated one at a time.
+         */
+
+
+        // Part 1: Parameters which do not have row scaling attached.
+        {
```

Comment:
Unless there is a particular need for blocks, i.e. reuse of variable names, I would rather than blocks, just use comments or functions to separate code, as too much indentation becomes hard to read.

Reply:
I have removed the blocks; my judgement is that it will be too low level to extract in a separate function - but I can do it.